snooty/parser.py

```diff
@@ -591,29 +593,39 @@ def handle_directive(
             openapi_fileid, filepath = util.reroot_path(
                 FileId(argument_text), self.docpath, self.project_config.source_path
             )
+            swagger_parse = True if "swagger-parse" in options else False
 
             try:
-                with open(filepath) as f:
-                    openapi = OpenAPI.load(f)
+                # swagger-parse leaves parsing to the Swagger React library in the frontend
+                if swagger_parse:
+                    with open(filepath) as f:
+                        spec = json.dumps(safe_load(f))
```

Comment:
Due to issues with MongoDB not allowing keys that begin with `$` to be stored, it was decided to store OpenAPI yaml file content as a JSON string instead of an object/dictionary.